### PR TITLE
add: noRoute and test

### PR DESCRIPTION
**Pull request by Ignorant-lxf**
I have some question.

I send a request for a exist route,it successed.Next I send a request for the noRoute，the allNoRoute(a attribute of THZ) may be changed by previous request,but the noRoute(a attribute of THZ) is invariable. I debug it  but not result.So I have to repeated invoke rebuild404Handlers when there is noRoute.

### CODE REVIEW

File: thz.go [hunk 1]
```diff
@@ -122,3 +130,20 @@ func (thz *THz) handle() func(c *fasthttp.RequestCtx) {
 		thz.ctxPool.Put(ctx)
 	}
 }
+
+func (thz *THz) NoRoute(handlers ...Handler) {
+	thz.noRoute = handlers
+	thz.rebuild404Handlers()
```
Comment (Ignorant-lxf) [inline thz.go:136]: Repeated invoke rebuild404Handlers

File: thz.go [hunk 2]
```diff
@@ -20,6 +20,9 @@ type THz struct {
 	route     *httprouter.Router[Context]
 	intercept Handlers
 
+	noRoute    Handlers
```
Comment (Ignorant-lxf) [inline thz.go]: It's invariable

File: thz.go [hunk 3]
```diff
@@ -20,6 +20,9 @@ type THz struct {
...
+	allNoRoute Handlers
```
Comment (Ignorant-lxf) [inline thz.go]: The allNoRoute may be changed and I don't know why.If you know,please tell me why.

**Comment by Ignorant-lxf** [comment]
https://github.com/X2OX/THz/issues/7

File: thz.go [hunk 4]
```diff
@@ -113,6 +116,11 @@ func (thz *THz) handle() func(c *fasthttp.RequestCtx) {
 			ctx.handlers = append(ctx.handlers, Handler(v))
 		}
 
+		if len(handlers) < 1 {
+			ctx.handlers = thz.allNoRoute
+			thz.rebuild404Handlers()
```
Comment (Ignorant-lxf) [inline thz.go]: Repeated invoke rebuild404Handlers

**Comment by Aoang** [comment]
It doesn't need to be that complicated.

https://github.com/X2OX/THz/blob/b40047ce5fb437e85ea1681a3a962505dbed7f39/thz.go#L111

If `handlers` is nil, append `THz.noRoute` to `ctx.handlers`.

**Pull request merged**